src/data/simple_dmatrix.cc

```diff
@@ -197,5 +197,9 @@ template SimpleDMatrix::SimpleDMatrix(FileAdapter* adapter, float missing,
                                      int nthread);
 template SimpleDMatrix::SimpleDMatrix(IteratorAdapter* adapter, float missing,
                                      int nthread);
+#if defined(XGBOOST_BUILD_ARROW_SUPPORT)
+template SimpleDMatrix::SimpleDMatrix(ArrowAdapter* adapter, float missing,
+                                     int nthread);
+#endif
```

Comment:
Overall, it will be good to add tests for Apache Arrow support on C++/Python side.

Reply:
Will add tests.